### ISSUE TITLE
Circular embeds are crashing Athens

### ISSUE THREAD

**Issue by snird** [athensresearch/athens#1125]
lets say I have this:
```
- something here
```

then I reference it in a bullet indented:
```
- something here
  - (reference to "something here")
```

Athens stops responding.
Of course, it doesn't make sense to use the software this way, but it happened to me by mistake and might happen to others

**Comment by sawhney17** [comment]
Doesn't stop reponding for me, what version of Athens are you on and what OS are you on?

**Comment by snird** [comment]
beta76 on macos

**Comment by yifanyin** [comment]
Can confirm. Embedding its parent block in a nested block causes Athens to freeze. macOS beta78

**Comment by juniusfree** [comment]
Adding here for reference. Might be related issues. https://github.com/athensresearch/athens/issues/1053

**Comment by sid597** [comment]
Cannot reproduce, everything works as expected beta.89. @snird @yifanyin can you please check this again in the ~latest builds?

**Comment by yifanyin** [comment]
Yup Athens still freezes when embedding parent block and CPU usage shoots up at the same time when parsing start. MacOS beta.89
